File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        ArrayList<Integer> list = new ArrayList<>();
        int n = nums.length;

        for(int i=0; i<n; i++) {
            if(nums[i] != val) {
                list.add(nums[i]);
            }
        }

        int ans = list.size();

        for(int i=0; i<ans; i++) {
            nums[i] = list.get(i);
        }

        return ans;
    }
}

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int i = 0;
        int in = 0;

        while(i < n) {
            if(nums[i] != val) {
                nums[in] = nums[i];
                i++;
                in++;
            } else {
                i++;
            }
        }
        return in;
    }
};